File: common/src/main/java/org/valkyrienskies/clockwork/util/blocktype/IHeatableBlock.java
package org.valkyrienskies.clockwork.util.blocktype;

import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.block.state.properties.EnumProperty;

public interface IHeatableBlock {

    public float heat = 0.0f;

    public static final EnumProperty<EngineHeatLevel> HEAT_LEVEL = EnumProperty.create("heat", EngineHeatLevel.class);

    static EngineHeatLevel getHeatLevelOf(BlockState blockState) {
        return blockState.hasProperty(HEAT_LEVEL) ? blockState.getValue(HEAT_LEVEL)
                : EngineHeatLevel.SMOULDERING;
    }



}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/content/logistics/gas/IHeatableBlock.kt
package org.valkyrienskies.clockwork.content.logistics.gas

import com.simibubi.create.content.equipment.goggles.IHaveGoggleInformation
import net.minecraft.network.chat.Component
import net.minecraft.world.level.block.state.properties.EnumProperty
import org.valkyrienskies.clockwork.ClockworkMod
import org.valkyrienskies.clockwork.kelvin.api.DuctNodePos
import org.valkyrienskies.mod.common.util.toJOMLD

interface IHeatableBlock {
    companion object {
        val GAS_HEAT_LEVEL: EnumProperty<GasHeatLevel> = EnumProperty.create("gas_heat_level", GasHeatLevel::class.java)
    }
}